### PR TITLE
V3.0 changes

### PR DESCRIPTION
**Pull request by ironmonk108**
- [x] Set minimum width/height
- [x] Add protection against hierarchy not being loaded.
- [x] Change module version
- [x] Need to figure out the Basic Rules issue with Items folders.  It's not putting the items in the Basic Rules folder... but recognizes that it's in the Basic Rules folder because deleting the main folder deletes all the items.
- [x] Removing debugging log messages
- [ ] Investigate Icon migration
- [ ] Single import of scene is not linking tokens or notes properly.
- [ ] Should be some sort of progress when creating documents
- [ ] v9 is not linking Actor links in Journal Entries.

### CODE REVIEW

File: src/hierarchy.js [hunk 1]
```diff
@@ -188,33 +187,41 @@ export class Hierarchy {
...
-                        part = { name: part };
+                if (path instanceof Array) {
+                    // traverse the path to make sure all folders are present
+                    for (let i = 0; i < path.length; i++) {
+                        let part = path[i];
+                        if (typeof part == "string") {
+                            part = { name: part };
+                        }
+                        // If the path is the actual entity id, then we've gone too far.
+                        if (document._id == part.id)
```
[the remark this answers is not in the document]
Comment (ironmonk88) [reply src/hierarchy.js:205]: Because at that point document is just a json object and not an actual Document.
So id doesn't exist, only _id does.

**Comment by rikmarais** [comment]
Great work, it's looking really good so far! I'm working my way through this. 

Making notes of any bugs I find in testing (that aren't `dndconverter` related): 

[ ] Using the `Import Book` feature to create scenes fails to link the tokens in those scenes to actors.
image
image

**Comment by kakaroto** [comment]
I assume that journal notes would also likely not be relinked to the appropriate journals when a scene is imported.
the "Journal Notes" folder is not really meant to be viewed by the users, just a place to store all the journal notes, because Atropos refused to give us the ability to link journal notes to a particular header in a journal 😬 

You should be able to match which journal/actor a note/token is linking to by matching the ID from the compendium (rather than trying to guess based on the name, etc..)

**Comment by rikmarais** [comment]
> I assume that journal notes would also likely not be relinked to the appropriate journals when a scene is imported. the "Journal Notes" folder is not really meant to be viewed by the users, just a place to store all the journal notes, because we can't link journal notes to a particular header in a journal.

The Journal Notes actually imported fine into their linked scenes. We had a brief chat on Discord and I think that IronMonk is going to hide them inside of the Compendium Library Journals tab. They're completely unsorted and you mentioned that they aren't really meant to be viewed by the user so I think that's good. That will possibly resolve the scrolling performance issue we observed

File: module.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "forge-compendium-browser",
   "title": "The Forge: Compendium Library",
   "description": "Browse your compendium books like they were meant to be viewed. The Forge's Compendium Library lets you navigate your D&amp;D Beyond imported books in a beautiful and easy to use interface!",
-  "version": "1.2.2",
+  "version": "2.0",
```
Comment (kakaroto) [inline module.json:5]: Don't switch from a `major.minor.patch` (semver) versioning system to a `major.minor` versioning system.

**Comment by kakaroto** [comment]
Not critical, but I also see that the scene info has shifted to the left instead of being centered
image

**Comment by ironmonk88** [comment]
That was on purpose.  I found with it displayed in the center a good part of the map was blocked.  And usually there's map you want to see in the center.  But the bottom left corner is mostly empty, or has the legend down there.

I had intentions to allow a user to use the wheel to zoom in/out and drag to move it around.  But I figured that could wait.

**Comment by ironmonk88** [comment]
Updated it to look like this.
image
Which I think makes it a lot easier to see the map, and still get the information across.  You can hover over the icons to see the text version.

**Comment by kakaroto** [comment]
> Updated it to look like this. image Which I think makes it a lot easier to see the map, and still get the information across. You can hover over the icons to see the text version.

It does look quite a bit nicer, thanks 👍 
Now please fix the folder thing as discussed (pseudocode) : 
```js
if (folder.depth > 3) {
   folder.parent = folder.parent.parent;
   folder.label = `${folder.parent.label}/${folder.label}`
}
```
And get this release out ASAP on monday. Thanks

**Pull request merged**